src/lightning_app/runners/cloud.py

```diff
@@ -224,6 +226,16 @@ def dispatch(
             repo.package()
             repo.upload(url=lightning_app_release.source_upload_url)
 
+            # check if user has sufficient credits to run an app
+            # if so set the desired state to running otherwise, create the app in stopped state,
+            # and open the admin ui to add credits and running the app.
+            has_sufficient_credits = self._project_has_sufficient_credits(project, app=self.app)
+            app_release_desired_state = (
+                V1LightningappInstanceState.RUNNING if has_sufficient_credits else V1LightningappInstanceState.STOPPED
+            )
+            if not has_sufficient_credits:
+                logger.info("You may need Lightning credits to run your apps on the cloud.")
```

Comment:
could be at a warning level

Also, why is it uncertain whether the user needs credits? Is it because we can't know the cloud compute being used in advance?

Reply:
yeah, dynamic works 🥲

Reply:
That's ok! Just wanted to know :)